### PR TITLE
Enable full HTML check after ajax load #4359

### LINKED ISSUE #4359: Enable full HTML check after ajax load
Currently, our `verifyHtmlAjax` is always followed by `verifyHtmlMainContent`. This gives _no_ platform if we want to do a full page check after ajax load.
As such, at least two of our pages have no full page check: `InstructorStudentList` and `InstructorStudentRecords`. There might be more, but I can say for sure for these two.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #4359 
This PR (and issue) will first add the capability for full HTML check after ajax load. The missing page checks are to be added in another (few) issue(s).

### CODE REVIEW

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 1]
```diff
@@ -888,6 +888,14 @@ public AppPage verifyHtmlMainContent(String filePath) {
         return this;
     }
     
+    public AppPage verifyHtmlAjaxMainContent(String filePath) throws Exception {
```
Comment (kanghj) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: It's a public method, so may be better to add some javadoc comments to this for easier reference in future.

**Comment by wkurniawan07** [comment]
Changes made. One last review, perhaps?

**Comment by kanghj** [comment]
Alright, looks good:)

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 2]
```diff
@@ -889,15 +889,35 @@ public AppPage verifyHtmlMainContent(String filePath) {
...
      * after "waitDuration", for "maxRetryCount" number of times.
      * @param filePath If this starts with "/" (e.g., "/expected.html"), the 
      * folder is assumed to be {@link Const.TEST_PAGES_FOLDER}. 
      * @return The page (for chaining method calls).
      */
     public AppPage verifyHtmlAjax(String filePath) throws Exception {
+        return verifyHtmlAfterAjaxLoad(filePath, true);
+    }
+
+    private AppPage verifyHtmlAfterAjaxLoad(String filePath, boolean isFullPageChecked) throws Exception {
```
Comment (thyageshm) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: hi, if possible change the second param to "isHtmlPart" as many of the functions that differentiate the two take that as the param. This is just for standardisation and to remove confusion

**Pull request merged**